docs-examples/2d/javascript/src/snippets/scene_queries.ts

```diff
@@ -0,0 +1,124 @@
+    // DOCUSAURUS: Shapecast start
+    let shapePos = { x: 0.0, y: 1.0 };
+    let shapeRot = 0.2;
+    let shapeVel = { x: 0.1, y: 0.4 };
+    let shape = new RAPIER.Cuboid(1.0, 2.0);
+    let targetDistance = 0.0;
+    let maxToi = 4.0;
+    // Optional parameters:
+    let stopAtPenetration = true;
+    let filterFlags = QueryFilterFlags.EXCLUDE_DYNAMIC;
+    let filterGroups = 0x000b0001;
+    let filterExcludeCollider = collider;
+    let filterExcludeRigidBody = rigidBody;
+
+    let hit = world.castShape(shapePos, shapeRot, shapeVel, shape, targetDistance, maxToi,
+        stopAtPenetration, filterFlags, filterGroups, filterExcludeCollider, filterExcludeRigidBody);
+    if (hit != null) {
+        // The first collider hit has the handle `handle`. The `hit` is a
+        // structure containing details about the hit configuration.
+        console.log("Hit the collider", hit.collider, "at time", hit.time_of_impact);
+    }
+    // DOCUSAURUS: Shapecast stop
+}
+
+{
+    // DOCUSAURUS: PointProjection start
+    let point = { x: 1.0, y: 2.0 };
+    let solid = true;
+
+    let proj = world.projectPoint(point, solid);
+    if (proj != null) {
+        // The collider closest to the point has this `handle`.
+        console.log("Projected point on collider ", proj.collider, ". Point projection: ", proj.point);
+        console.log("Point was inside of the collider shape: {}", proj.isInside);
+    }
+
+    world.intersectionsWithPoint(point, (handle) => {
+        // Callback called on each collider with a shape containing the point.
+        console.log("The collider", handle, "contains the point.");
+        // Return `false` instead if we want to stop searching for other colliders containing this point.
+        return true;
+    });
+    // DOCUSAURUS: PointProjection stop
+}
+{
+    // DOCUSAURUS: IntersectionTest start
+    let shape = new RAPIER.Cuboid(1.0, 2.0);
+    let shapePos = { x: 1.0, y: 2.0 };
+    let shapeRot = 0.1;
+
+    world.intersectionsWithShape(shapePos, shapeRot, shape, (handle) => {
+        console.log("The collider", handle, "intersects our shape.");
+        return true; // Return `false` instead if we want to stop searching for other colliders that contain this point.
+    });
+
+    let aabbCenter = { x: -1.0, y: -2.0 };
+    let aabbHalfExtents = { x: 0.5, y: 0.6 };
+    world.collidersWithAabbIntersectingAabb(aabbCenter, aabbHalfExtents, (handle) => {
+        console.log("The collider", handle, "has an AABB intersecting our test AABB");
+        return true; // Return `false` instead if we want to stop searching for other colliders that contain this point.
+    });
+    // DOCUSAURUS: IntersectionTest stop
+}
+{
+    let rigidBodyDesc = RAPIER.RigidBodyDesc.dynamic();
+    let player_rigid_body = world.createRigidBody(rigidBodyDesc);
+    let collider = world.createCollider(new RAPIER.ColliderDesc(new RAPIER.Ball(0.5)));
+    let data = new Map();
+    data.set(collider.handle, 10);
```

Comment:
I'm not sure about how user would store user data in a collider, I imagine in javascript they could add whatever they want on the type, but in typescript I believe there's no other choice than adding it externally ?